azkustoingest/query_client.go

```diff
@@ -0,0 +1,18 @@
+package azkustoingest
```

Comment:
Why is this public interface part of azkustoingest

Reply:
This is one of the diffs between go and other packages.

In most Sdks, you create an ingest client and it creates a query client within it

In go, you give it a made client in the ctor:
`
azkustoingest.New(azkustodata.New("blabla"))
`

The interface is here because it's what ingest needs, not what azkustodata provides if it makes sense.

Maybe we should take the chance at this point and make it work like the rest of the SDKs, but this better be in another PR.